tensorflow_model_optimization/python/core/common/keras/compression/algorithms/BUILD

```diff
@@ -43,3 +46,27 @@ py_test(
         # tensorflow dep1,
     ],
 )
+
+py_library(
+    name = "weight_clustering",
+    srcs = ["weight_clustering.py"],
+    srcs_version = "PY3",
+    deps = [
+        # tensorflow dep1,
+        "//tensorflow_model_optimization/python/core/clustering/keras:clustering_centroids",
```

Comment:
Let's have someone from Arm look at this PR given this added dependency.

It'd probably worth copying some aspects of the high level doc (not the design, but just the goals / motivation) and have thing as a pending PR, for communication purposes.

Reply:
Copied some portions of high level doc to https://github.com/tensorflow/model-optimization/pull/581.

Reply:
Is it better to add weight clustering tag for this PR?

Reply:
No need to add the tag. 

@akarmi, @wwwind  for visibility. See the above high level doc - this is a continuation from the discussions a long time back on having a shared internal API for different compression techniques.

Reply:
The 'weight compression' tag contains different PRs that are relevant and we're going to increasingly make things public.

Reply:
acknowledged.